### PR TITLE
Refactor config package and add bundle-definition tests

### PR DESCRIPTION
**Pull request by afoxman**
### Description

Refactor kit config types, pulling out bundler runtime parameters. This does not impact the package API, but is important for the upcoming Metro server work. 

Add tests for the bundle definition code. Move existing tests into their own directory, and update jest config and snapshots.

### Test plan

Verify that all tests pass, including the new tests.

### CODE REVIEW

File: packages/config/src/bundleConfig.ts [hunk 1]
```diff
@@ -74,13 +42,50 @@ export type BundleRequiredParameters = {
...
 
   /**
-   * Choose whether to enable tree shaking.
+    * Choose whether to enable tree shaking.
 
-   * ⚠️ **THIS FEATURE IS HIGHLY EXPERIMENTAL** ⚠️
+    * ⚠️ **THIS FEATURE IS HIGHLY EXPERIMENTAL** ⚠️
+    *
+    * @default false
+    */
```
Comment (tido64) [inline packages/config/src/bundleConfig.ts:50]: Something weird happened with the formatting here…

**Review by tido64 — COMMENTED** [review]
This looks fine to me, but it looks like you're waiting for #444? Or do you plan to clean up the change files and rebase?

**Comment by afoxman** [comment]
> This looks fine to me, but it looks like you're waiting for #444? Or do you plan to clean up the change files and rebase?

Yes, I will pull/merge after CI publishing (no rebasing), and this one will magically be ready for main.

File: packages/config/src/bundleConfig.ts (hunk 1 above)
Comment (afoxman) [reply packages/config/src/bundleConfig.ts]: Looks like extra spaces got in there somehow. Fixed.

**Review by JasonVMo — APPROVED** [review]
Glad to see the cleanup and additional tests

**Pull request merged**